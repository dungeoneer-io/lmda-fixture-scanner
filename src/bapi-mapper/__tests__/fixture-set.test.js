const { describe, expect, test } = require('@jest/globals');

const {
    convertCrealmIndexToMappingFixture,
    convertSpecIndexToSpecMap,
    convertDungeonIndexToDungeonMap
} = require('../fixture-set');

const ArrayOfDungeons = require('./__fixtures__/array-of-dungeons.json');
const ArrayOfSpecs = require('./__fixtures__/array-of-specs.json');
const ArrayOfCrealms = require('./__fixtures__/array-of-crealms.json');

describe('transformers', () => {
    test('convertCrealmIndexToMappingFixture', () => {
        const actual = convertCrealmIndexToMappingFixture(ArrayOfCrealms);
        expect(actual).toMatchSnapshot();
    });

    test('convertSpecIndexToSpecMap', () => {
        const result = convertSpecIndexToSpecMap(ArrayOfSpecs);
        expect(result).toMatchSnapshot();
    });

    test('convertDungeonIndexToDungeonMap', () => {
        const result = convertDungeonIndexToDungeonMap(ArrayOfDungeons);
        expect(result).toMatchSnapshot();
    });
});