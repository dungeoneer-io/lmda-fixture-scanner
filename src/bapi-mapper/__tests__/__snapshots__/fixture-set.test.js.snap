// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformers convertCrealmIndexToMappingFixture 1`] = `
{
  "10": 121,
  "110": 127,
  "1143": 121,
  "1148": 127,
  "1182": 121,
  "121": 121,
  "124": 127,
  "127": 127,
  "1348": 127,
  "1358": 127,
  "1359": 121,
  "1362": 127,
  "151": 151,
  "3": 151,
  "7": 127,
}
`;

exports[`transformers convertDungeonIndexToDungeonMap 1`] = `
{
  "169": [
    1800000,
    1440000,
    1080000,
  ],
}
`;

exports[`transformers convertSpecIndexToSpecMap 1`] = `
{
  "72": {
    "c": 1,
    "r": 3,
  },
  "73": {
    "c": 1,
    "r": 1,
  },
}
`;
